internal/awsdata/elasticache.go

```diff
@@ -0,0 +1,84 @@
+package awsdata
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/manywho/awsinventory/internal/inventory"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// AssetTypeElastiCacheNode is the value used in the AssetType field when fetching ElastiCache nodes
+	AssetTypeElastiCacheNode string = "ElastiCache Node"
+
+	// ServiceElastiCache is the key for the ElastiCache service
+	ServiceElastiCache string = "elasticache"
+)
+
+func (d *AWSData) loadElastiCacheNodes(region string) {
+	defer d.wg.Done()
+
+	elasticacheSvc := d.clients.GetElastiCacheClient(region)
+
+	log := d.log.WithFields(logrus.Fields{
+		"region":  region,
+		"service": ServiceElastiCache,
+	})
+	log.Info("loading data")
+	var cacheClusters []*elasticache.CacheCluster
+	done := false
+	params := &elasticache.DescribeCacheClustersInput{
+		ShowCacheNodeInfo: aws.Bool(true),
+	}
+	for !done {
+		out, err := elasticacheSvc.DescribeCacheClusters(params)
+		if err != nil {
+			d.results <- result{Err: err}
+			return
+		}
+
+		cacheClusters = append(cacheClusters, out.CacheClusters...)
+
+		if out.Marker == nil {
+			done = true
+		} else {
+			params.Marker = out.Marker
+		}
+	}
+
+	log.Info("processing data")
+	for _, c := range cacheClusters {
+		var vpcId string
+		groups, err := elasticacheSvc.DescribeCacheSubnetGroups(&elasticache.DescribeCacheSubnetGroupsInput{
+			CacheSubnetGroupName: c.CacheSubnetGroupName,
+		})
+		if err != nil {
+			d.results <- result{Err: err}
+		} else if len(groups.CacheSubnetGroups) > 0 {
+			vpcId = aws.StringValue(groups.CacheSubnetGroups[0].VpcId)
+		}
+
+		for _, n := range c.CacheNodes {
+			d.results <- result{
+				Row: inventory.Row{
+					UniqueAssetIdentifier:     fmt.Sprintf("%s-%s", aws.StringValue(c.CacheClusterId), aws.StringValue(n.CacheNodeId)),
+					Virtual:                   true,
+					Public:                    false,
+					DNSNameOrURL:              aws.StringValue(n.Endpoint.Address),
+					BaselineConfigurationName: aws.StringValue(c.CacheParameterGroup.CacheParameterGroupName),
+					Location:                  region,
+					AssetType:                 AssetTypeElastiCacheNode,
+					HardwareMakeModel:         aws.StringValue(c.CacheNodeType),
+					// TODO: SoftwareDatabaseVendor
```

Comment:
Do we know if c.Engine could be used as this value?

Reply:
Yeah, close enough. Redis was the one I was a bit concerned about, since it's technically "Redis Labs", but 🤷 .

Reply:
We could do something like this above and then use that variable?
```
vendor := c.Engine
if vendor == "redis" {
    vendor = "Redis Labs"
}
``

Reply:
https://en.wikipedia.org/wiki/Redis#History makes it seem like it is "sponsored" by Redis Labs, versus owned. The more correct thing is probably just saying "Salvatore Sanfilippo" (the actual author), but I think just leaving it as "redis" is fine, imho, especially as being an open source project. Thoughts?

Reply:
Works for me!